handlers.go

```diff
@@ -81,3 +81,40 @@ func checksHandler(c *echo.Context) error {
 
 	return c.JSON(http.StatusOK, checks)
 }
+
+// DELETE /sites/:key
+func deleteSiteHandler(c *echo.Context) error {
+	siteKey := c.P(0)
+
+	_, err := db.Exec("DELETE FROM sites WHERE id=$1", siteKey)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, "Error deleting site")
+	}
+
+	_, err = db.Exec("DELETE FROM checks WHERE siteId=$1", siteKey)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, "Error deleting site checks")
+	}
+
+	removeSite(siteKey)
```

Comment:
the `removeSite` function is what should do the above `DELETE` statements and return an `error`.  The `deleteSiteHandler` can just do

    err := removeSite(siteKey)
    if err != nil {
      ...
    }